src/main/java/org/terasology/changingBlocks/ChangingBlocksSystem.java

```diff
@@ -94,8 +83,8 @@ public void update(float delta) {
                     LocationComponent locComponent = changingBlocks.getComponent(LocationComponent.class);
                     Block currentBlock = worldprovider.getBlock(locComponent.getWorldPosition(new Vector3f()));
                     String currentBlockFamilyStage = currentBlock.getURI().toString();
-                    Set<String> keySet = blockAnimation.blockFamilyStages.keySet();
-                    List<String> keyList = new ArrayList<>(keySet);
+                    Set<SimpleUri> keySet = blockAnimation.blockFamilyStages.keySet();
+                    List<SimpleUri> keyList = new ArrayList<>(keySet);
                     int currentstageIndex = keyList.indexOf(currentBlockFamilyStage);
```

Comment:
The SpotBugs warning below should be taken serious here, as it is an easy to miss error when Java compare random object (no compile errors).

Reply:
done